### ISSUE TITLE
Cannot call pretrained PINN models

### ISSUE THREAD

**Issue by Ognjen303** [pdebench/PDEBench#42]
Hi,

First, thank you for the great work you have done.

I am trying to use your pretrained models to get the exact same results as from your paper but cannot figure out how to do so. I am able to train my own PINNs, but not to use the pretrained PINNs.

For example, I am trying to run 1D_Advection_Sols_beta4.0_PINN.pt-15000.pt (available [here](https://darus.uni-stuttgart.de/file.xhtml?fileId=133595&version=1.0) once downloaded and unzipped). This pretrained model should give the numbers as in the last column for beta=4.0 of page 13 of the Appendix. But I cannot figure out how to run this pretrained model.

In contrast, for example, when using a pretrained Unet, I can do so by setting continue_training = True in the config.yaml file. But I don't see any such functionality for PINN.

Your help would be very much appreciated.

All the best,
Ognjen Stefanovic

**Comment by kmario23** [comment]
Hey @Ognjen303, Hvala!
Such a functionality is not yet implemented, but it would be nice to have for verification purposes. Would you be interested in sending a PR?

Here are some tips:

-  remove/disable [`model.train(...)`](https://github.com/pdebench/PDEBench/blob/main/pdebench/models/pinn/train.py#L406)
-  *restore* the model `model.restore("path/to/checkpoint", ...)` after `model.compile()`
- call [`model.predict(...)` on the test data ](https://github.com/pdebench/PDEBench/blob/main/pdebench/models/pinn/train.py#L415)

For details when implementing, see [deexde general usage Q&A](https://deepxde.readthedocs.io/en/latest/user/faq.html#general-usage)
> Q: How can I save a trained model and then load the model later?
A: [#54](https://github.com/lululxvi/deepxde/issues/54), [#57](https://github.com/lululxvi/deepxde/issues/57), [#58](https://github.com/lululxvi/deepxde/issues/58), [#63](https://github.com/lululxvi/deepxde/issues/63), [#103](https://github.com/lululxvi/deepxde/issues/103), [#206](https://github.com/lululxvi/deepxde/issues/206), [#254](https://github.com/lululxvi/deepxde/issues/254), [#468](https://github.com/lululxvi/deepxde/issues/468)

**Comment by camolina2** [comment]
Hi @Ognjen303 and @kmario23, I am also having the same problem. Even though I can save the model properly, I cannot load it to make new predictions.

I have already tried to read almost all the issues. If you have any advice or if you have solved the problem, I would appreciate your help.

**Comment by leiterrl** [comment]
Hi @camolina2 ,

I just successfully loaded the pre-trained weights by using `model.restore` as @kmario23 suggested. I think the main culprit here is the option to use different backends in deepxde. Your comment suggests that you are trying to use the old tensorflow1 backend. Note that the PDEBench models were trained using the _PyTorch_ backend. Please see the deepxde documentation on how to switch to a specific backend: https://deepxde.readthedocs.io/en/latest/user/installation.html#working-with-different-backends

Best,
Raphael

**Comment by camolina2** [comment]
I managed to load the trained network using model.train(), but it doesn't work with model.restore(). I cannot find the reason.

Anyway, if I reload it using model.train(), I have to define some epochs. I was putting epochs=0 because I don't want to train the network again; I just want to load it. However, I am not getting the same results in the loss function as before. Do you have any idea of the reason?

**Comment by leiterrl** [comment]
According to the deepxde issues mentioned above you have to specify the number of epochs including the ones already pre-trained. So for example when loading `1D_Advection_Sols_beta4.0_PINN.pt-15000.pt` you would have to specify `epochs=15000` in your call to `model.train`.
